cloud/scope/powervs_machine_test.go

```diff
@@ -190,6 +327,764 @@ func TestNewPowerVSMachineScope(t *testing.T) {
+					},
+				},
+				Machine: &capiv1beta1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							tc.machineLabel: "",
+						},
+					},
+				},
+			}
+			g.Expect(tc.expectedBootstrapDataKey).To(Equal(machineScope.bootstrapDataKey()))
+		})
+	}
+}
+
+func TestGetNetworkID(t *testing.T) {
+	var (
+		mockCtrl *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	const networkID = "foo-network-id"
+	t.Run("Get Network ID", func(t *testing.T) {
+		t.Run("Fetch from Network spec's ID", func(t *testing.T) {
+			scope := PowerVSMachineScope{}
+			expectedNetworkID := networkID
+			networkResource := infrav1beta2.IBMPowerVSResourceReference{
+				ID: core.StringPtr(expectedNetworkID),
+			}
+			result, err := getNetworkID(networkResource, &scope)
+			require.Nil(t, err)
+			require.Equal(t, *result, expectedNetworkID)
+		})
+		t.Run("Fetch network ID from PowerVS Machine scope", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			networkName := "foo-network-name"
+			expectedNetworkID := networkID
+			networkReferences := []*models.NetworkReference{
+				{
+					Name:      core.StringPtr(networkName),
+					NetworkID: core.StringPtr(expectedNetworkID),
+				},
+			}
+			networks := &models.Networks{
+				Networks: networkReferences,
+			}
+			networkResource := infrav1beta2.IBMPowerVSResourceReference{
+				Name: core.StringPtr(networkName),
+			}
+
+			mockResourceController := mock.NewMockPowerVS(gomock.NewController(t))
+			mockResourceController.EXPECT().GetAllNetwork().Return(networks, nil)
+			scope := PowerVSMachineScope{
+				IBMPowerVSClient: mockResourceController,
+			}
+			result, err := getNetworkID(networkResource, &scope)
+			require.Nil(t, err)
+			require.Equal(t, *result, expectedNetworkID)
+		})
+
+		t.Run("Failed to find network ID", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			expectedNetworkID := networkID
+			differentNetworkID := "diff-network-id"
+			networkReferences := []*models.NetworkReference{
+				{
+					Name: core.StringPtr(differentNetworkID),
+				},
+			}
+			networks := &models.Networks{
+				Networks: networkReferences,
+			}
+			networkResource := infrav1beta2.IBMPowerVSResourceReference{
+				Name: core.StringPtr(expectedNetworkID),
+			}
+
+			mockResourceController := mock.NewMockPowerVS(gomock.NewController(t))
+			mockResourceController.EXPECT().GetAllNetwork().Return(networks, nil)
+			scope := PowerVSMachineScope{
+				IBMPowerVSClient: mockResourceController,
+			}
+			result, err := getNetworkID(networkResource, &scope)
+			require.Nil(t, result)
+			require.EqualError(t, err, fmt.Sprintf("failed to find a network ID with name %s", expectedNetworkID))
+		})
+
+		t.Run("Fetch network ID with matching regex", func(t *testing.T) {
+			setup(t)
+			t.Cleanup(teardown)
+			networkName := "550e8400-e29b-41d4-a716-446655440000"
```

Comment:
Is it intentional for assigning id to name?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/cloud/scope/powervs_machine.go#L703 - here the regex is checked against networkName